lib/cfg-block-generator.c

```diff
@@ -39,12 +39,17 @@ _report_generator_args(gpointer key, gpointer value, gpointer user_data)
 {
   GString *result = (GString *) user_data;
   g_string_append_printf(result, "## %s=", (gchar *) key);
-  for (const gchar *c = (const gchar *) value; *c; c++)
+  if (value != NULL)
     {
-      if (*c == '\n' && *(c + 1))
-        g_string_append(result, "\n## ");
-      else
-        g_string_append_c(result, *c);
+      for(const gchar *c = (const gchar *) value;
```

Comment:
please keep the original expression format here to stay consistent with the other usages
also, I guess a simple

`for (const gchar *c = (const gchar *) value; c && *c; c++)`

could do the trick

Reply:
`c && *c` would, in my opinion be less readable than an `if` statement but it should compile identically, so I could change it if you want. The formatting should be fixed, I'll do that regardless.

Reply:
My bad. A null-check in a loop would be slower than a null-check once.